python/ray/data/dataset.py

```diff
@@ -1180,6 +1180,11 @@ def filter(
         stateful Ray actors. For more information, see
         :ref:`Stateful Transforms <stateful_transforms>`.
 
+        .. tip::
+           If you can represent your filter as an expression that leverages Arrow
+           Dataset Expression, we will be do highly optimized filtering using native
+           Arrow interfaces.
+
```

Comment:
Let's remove the two below tips?

> If you can represent your predicate with NumPy or pandas operations,
            :meth:`Dataset.map_batches` might be faster. You can implement filter by
            dropping rows.

>  If you're reading parquet files with :meth:`ray.data.read_parquet`,
            and the filter is a simple predicate, you might
            be able to speed it up by using filter pushdown; see
            :ref:`Parquet row pruning <parquet_row_pruning>` for details.

Reply:
And for "Parquet row pruning", let's remove the corresponding section from the performance tips user guide?

Reply:
That seemed useful to me when I saw the tip and it's valid even now. Let me remove it and upload here. Will help with discussion if it's still valid.